src/ring_theory/integral_closure.lean

```diff
@@ -27,23 +27,40 @@ open_locale big_operators
 open polynomial submodule
 
 section ring
-variables (R : Type*) {A : Type*}
+variables {R A : Type*}
 variables [comm_ring R] [ring A]
-variables [algebra R A]
+
+/-- An element `x` of `A` is said to be integral over `R` with respect to `f`
+if it is a root of a monic polynomial `p : polynomial R` evaluated under `f` -/
+def is_integral (f : R →+* A) (x : A) :=
+∃ p : polynomial R, monic p ∧ eval₂ f x p = 0
+
+/-- A ring homomorphism `f : R →+* A` is said to be integral
+if every element `A` is integral with respect to the map `f` -/
+def ring_hom.is_integral (f : R →+* A) :=
+∀ x : A, is_integral f x
+
+variables [algebra R A] (R)
 
 /-- An element `x` of an algebra `A` over a commutative ring `R` is said to be *integral*,
-if it is a root of some monic polynomial `p : polynomial R`. -/
-def is_integral (x : A) : Prop :=
-∃ p : polynomial R, monic p ∧ aeval x p = 0
+if it is a root of some monic polynomial `p : polynomial R`.
+Equivalently, the element is integral over `R` with respect to the induced `algebra_map` -/
+def is_integral_alg (x : A) : Prop :=
```

Comment:
And then this my stay the way it was:
```suggestion
def is_integral (x : A) : Prop :=
```

Reply:
Switched back, always a positive not to break unmerged branches I suppose